### ISSUE TITLE
Need to handle 429 errors (Rate Limit throttled)

### ISSUE THREAD

**Issue by jakerator** [saleweaver/python-amazon-sp-api#24]
**Which API - Endpoint is your request about.**

All API endpoints

**Additional context**

Here is how it any API call fails when throttle limit reached:
------> sp_api.base.client.SellingApiException: [{'message': 'You exceeded your quota for the requested resource.', 'code': 'QuotaExceeded'}]

Probably need to add some "re-trying" right in this library. Or maybe add an Exception handling...

**Comment by saleweaver** [comment]
I thought retrying directly from the library is a bad idea and deciding what should happen for any error occurred should be left to the consumer. There are so many ways on how to handle this error and a wrapper shouldn't be responsible to decide for the consumer (IMHO). It should be fairly simple to add error handling on your end though

``` 
try:
    Orders()...
except SellingApiException as sae:
    # check the error code and do what you want with it. e.g. retry the call in 5 minutes.

**Comment by jakerator** [comment]
> try:
>     Orders()...
> except SellingApiException as sae:
>     # check the error code and do what you want with it. e.g. retry the call in 5 minutes.
> ```
Pretty difficult to fetch exception code from SellingApiException . Maybe need to add another one for RateThrottled or add track SP API error code in SellingApiException somewhere...

**Comment by saleweaver** [comment]
Yeah, that makes sense. I'll add that later

**Comment by saleweaver** [comment]
Error codes and multiple exceptions available in v0.1.7
